pkg/scheduler/util/utils.go

```diff
@@ -63,6 +65,31 @@ func GetPodPriority(pod *v1.Pod) int32 {
 	return scheduling.DefaultPriorityWhenNoDefaultClassExists
 }
 
+// GetEarliestPodStartTime returns the earliest start time of all pods that
+// have the highest priority among all victims.
+func GetEarliestPodStartTime(victims *api.Victims) *metav1.Time {
+
+	if len(victims.Pods) == 0 {
+		// should not reach here. add this for completeness
+		return nil
+	}
+
+	earliestPodStartTime := victims.Pods[0].Status.StartTime
+	highestPriority := GetPodPriority(victims.Pods[0])
+
+	for _, pod := range victims.Pods {
+		if *pod.Spec.Priority < highestPriority {
+			// break since the victims are ranked from highest priority to lowest
```

Comment:
This is brittle. Your logic in this function makes a strong assumption about the order of victims which is implemented in another function.
Since this function is not called frequently and we don't expect the number of victims to be large (often times less than 10), I don't think it is worth optimizing the logic at the expense of making it error prone.
I would remove the `break` and just compare priority and if it is greater than or equal to the highest priority then update startup time if needed.